src/main/java/com/worksap/nlp/sudachi/Dictionary.java

```diff
@@ -72,17 +72,33 @@ public interface Dictionary extends AutoCloseable {
     /**
      * Lookup entries in the dictionary without performing an analysis.
      * 
-     * Specified surface will be normalized. This will work like performing analysis
-     * on the given headword and find paths with a single morpheme, but returns all
+     * Specified surface will be normalized. This works like performing analysis on
+     * the given headword and find paths with a single morpheme, but returns all
      * paths instead of the lowest cost one.
      * 
      * @param surface
-     *            to lookup. Will be normalized beforehand.
+     *            surface to lookup. Will be normalized beforehand.
      * @return a list of morphemes that match the surface. Their begin/end will be
      *         0/length of their headword.
      */
     public List<Morpheme> lookup(CharSequence surface);
 
+    /**
+     * Lookup from all entries in the dictionary.
+     * 
+     * Specified surface will be normalized. This can find entries that are not
+     * indexed and appear only when referred from other words (e.g. constitution),
+     * but is VERY slow instead. {@link Dictionary#lookup(CharSequence)} should be
+     * used for most cases.
+     * 
+     * @param surface
+     *            surface to lookup. Will be normalized beforehand.
+     * @return a list of morphemes that match the surface. Their begin/end will be
+     *         0/length of their headword.
+     * @see Dictionary#lookup(CharSequence)
+     */
+    public List<Morpheme> slowLookupAllEntries(CharSequence surface);
```

Comment:
I don't think it is necessary to include `slow` in the name, since slow is an implementation issue, not a feature to be provided.